lib/launcher.js

```diff
@@ -47,13 +47,18 @@ class Launcher {
             return reporter
         }
 
-        for (let reporterType of config.reporter) {
-            try {
-                let Reporter = require(`wdio-${reporterType}-reporter`)
-                reporter.add(new Reporter(reporter, config))
-            } catch (e) {
-                throw new Error(`reporter "wdio-${reporterType}-reporter" is not installed. Error: ${e.stack}`)
+        for (let Reporter of config.reporter) {
+            if (typeof Reporter === 'string') {
+                try {
+                    Reporter = require(`wdio-${Reporter}-reporter`)
+                } catch (e) {
+                    throw new Error(`reporter "wdio-${Reporter}-reporter" is not installed. Error: ${e.stack}`)
+                }
             }
+            if (typeof Reporter !== 'function') {
+                throw new Error(`reporter option must be either string or function, but got: ${Reporter}`)
```

Comment:
Usually, most values are converted to string as `[object Object]` and I considered that is would be good to explain the error.